testsuite/src/it/java/org/graalvm/tests/integration/utils/Apps.java

```diff
@@ -178,34 +181,31 @@ public enum Apps {
         this.whitelistLogLines = whitelistLogLines;
         this.buildAndRunCmds = buildAndRunCmds;
         this.runtimeContainer = runtimeContainer;
-        File tpFile = new File(BASE_DIR + File.separator + dir + File.separator + "threshold.properties");
-        // Some apps don't have threshold.properties
-        if (tpFile.exists()) {
-            String appDirNormalized = dir.toUpperCase()
-                    .replace(File.separator, "_")
-                    .replace('-', '_')
-                    + "_";
-            try (InputStream input = new FileInputStream(tpFile)) {
-                Properties props = new Properties();
-                props.load(input);
-                for (String pn : props.stringPropertyNames()) {
-                    String normPn = pn.toUpperCase().replace('.', '_');
-                    String env = System.getenv().get(appDirNormalized + normPn);
+
+        // Some apps don't have threshold.conf
+        final Path tcFile = Path.of(BASE_DIR, dir, "threshold.conf");
+        if (Files.exists(tcFile)) {
+            final String appDirNormalized = dir.toUpperCase().replace(File.separator, "_").replace('-', '_') + "_";
+            try {
+                final Map<String, Long> props = parseProperties(tcFile);
+                for (String pn : props.keySet()) {
+                    final String normPn = pn.toUpperCase().replace('.', '_');
+                    final String env = System.getenv().get(appDirNormalized + normPn);
                     if (StringUtils.isNotBlank(env)) {
-                        props.replace(pn, env);
+                        props.replace(pn, Long.parseLong(env));
```

Comment:
Do we really want to keep the ability to override thresholds? What's the use case?

Reply:
I am not sure people even knew it was there.
I used to use it back in the day in internal Jenkins jobs where I overwrote certain threshold according to the particular executor (stronger vs. weaker vm) as a JOB env parameter.

I don't think it is being actively used anywhere now...

Reply:
Ha, it's been documented all along though: 

https://github.com/Karm/mandrel-integration-tests#thresholds